pkg/util/ready/ready.go

```diff
@@ -24,6 +27,26 @@ import (
 	apiregistrationv1client "k8s.io/kube-aggregator/pkg/client/clientset_generated/clientset/typed/apiregistration/v1"
 )
 
+// IsMatchingSyscallError returns true when the error is one of the Errno's in match
+// it deals with the different ways of wrapping the syscalls.
+func IsMatchingSyscallError(err error, match []syscall.Errno) bool {
+	switch t := err.(type) {
+	case *url.Error:
+		return IsMatchingSyscallError(t.Err, match)
+	case *net.OpError:
+		return IsMatchingSyscallError(t.Err, match)
+	case *os.SyscallError:
+		return IsMatchingSyscallError(t.Err, match)
+	case syscall.Errno:
+		for _, sc := range match {
+			if t == sc {
+				return true
+			}
+		}
+	}
```

Comment:
I'm on the fence about adding a `default:` case + panic in the case of an unrecognised type - wdyt?

Reply:
The point of the function is return true/false to "is this one of the syscall errors", so you can pass in
any error and you don't want to panic'ing, but rather returning false so that the caller can handle that
error differently.